components/org.wso2.carbon.appmgt.api/src/main/java/org/wso2/carbon/appmgt/api/APIConsumer.java

```diff
@@ -98,13 +98,22 @@
                                                                            AppManagementException;
 
     /**
-     * Get all tags of published APIs
+     * Get all tags of published Apps
      *
-     * @return a list of all Tags applied to all APIs published.
-     * @throws AppManagementException if failed to get All the tags
+     * @return a list of all tags applied to all apps published.
+     * @throws AppManagementException if failed to get all the tags
      */
     public Set<Tag> getAllTags(String tenantDomain) throws AppManagementException;
 
+
+    /**
+     * Get all tags of published Apps
+     *
+     * @return a list of all tags applied to all apps published.
+     * @throws AppManagementException if failed to get all the tags
+     */
+    public Set<Tag> getAllTags(String tenantDomain, String assetType, String assetFlag) throws AppManagementException;
```

Comment:
make the method more generic to get the tags based on any filtering criteria i.e instead of assetFlat, pass maps of attributes and filter based on that.